### PR TITLE
[WIP] Implementing docker-based HHVM+Nginx bootstrapper

### PR DESCRIPTION
**Pull request by Ocramius**
Todos:
- [x] create data containers (public dir)
- [x] share data container with the relevant docker containers 
- [ ] ~~create test runner containers~~
- [ ] ~~vagrant provisioning script (minimal)~~
- [ ] ~~upgrade to ubuntu 14.04 (because.)~~
- [ ] ~~...?~~

### CODE REVIEW

File: docker/NGINX/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# meh, this path has to be fixed
+ADD nginx.conf /etc/nginx/nginx.conf
+ADD nginx.sh nginx.sh
+ADD default.conf /etc/nginx/sites-available/default
+
+EXPOSE 80
+
+# USER www-data
+
+CMD sh nginx.sh
```
Comment (deeky666) [inline docker/NGINX/Dockerfile:22]: you don't need `sh` here again because `CMD` is infact a `sh -c` call

File: docker/NGINX/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+MAINTAINER Marco Pivetta, ocramius@gmail.com
+
+# Install Nginx
+
+#RUN echo "deb http://archive.ubuntu.com/ubuntu/ raring main universe" >> /etc/apt/sources.list
+
+RUN export DEBIAN_FRONTEND=noninteractive
+RUN apt-get update
+RUN apt-get install -y nano wget dialog net-tools
+RUN apt-get install -y nginx
```
Comment (deeky666) [inline docker/NGINX/Dockerfile]: You can put those install commands into one

File: docker/run.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+
+docker kill NGINX
+docker kill HHVM
+
+docker rm HHVM
+docker rm NGINX
```
Comment (deeky666) [inline docker/run.sh:7]: you can sum those 4 commands into `docker rm -f NGINX HHVM`. Be careful,  docker will error with `kill` and `rm` if the containers don't exist.

File: docker/run.sh [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+docker rmi hhvm
+docker rmi nginx
```
Comment (deeky666) [inline docker/run.sh:10]: Same here, error if not exists

File: docker/NGINX/Dockerfile (hunk 1 above)
Comment (Ocramius) [inline docker/NGINX/Dockerfile:22]: good to know, but it actually failed here before...

File: docker/run.sh (hunk 3 above)
Comment (Ocramius) [inline docker/run.sh:7]: Yeah, if it errors here it is no big deal tho. This is mainly "cleanup and redo"

**Comment by deeky666** [comment]
Awesome! As an improvement I would suggest making VOLUMES on both images for logs, conf, sites etc. if you would like that. So you can share those data easy between host and container :)

**Comment by Ocramius** [comment]
@deeky666 will surely do! Thanks for the kickstarter so far :-)

File: docker/NGINX/default.conf [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+    location ~ \.php$ {
+    #    proxy_set_header   X-Real-IP        $remote_addr;
+    #    proxy_set_header Host docker-nginx;
+    #    proxy_pass   %php.proto%://%php.host%:%php.port%;
+    #    proxy_pass   http//%php.host%:%php.port%;
+
+        fastcgi_keep_conn on;
+        fastcgi_pass   %php.host%:%php.port%;
+        fastcgi_index  index.php;
```
Comment (Ocramius) [inline docker/NGINX/default.conf:8]: Should also evaluate using `fastcgi.fix_pathinfo`

File: .gitignore [hunk 6]
```diff
@@ -1,5 +1,2 @@
-INSTALL.txt
-LICENSE.txt
-README.txt
-demos/
-extras/documentation
+vagrant/.vagrant
+vendor
```
Comment (cordoval) [inline .gitignore:2]: why the vendor folder here?

**Comment by cordoval** [comment]
also why not merging this PR instead of committing directly to branch? https://github.com/Ocramius/ZfKickstarter/commit/6f47ba62d3d5c79d72607edb7d0e4c35ca81a13d ?

File: .gitignore (hunk 6 above)
Comment (Ocramius) [inline .gitignore:2]: Vendor should be ignored in this project, but not globally

**Comment by Ocramius** [comment]
I merge a PR only once it's done.

File: docker/NGINX/default.conf [hunk 7]
```diff
@@ -0,0 +1,13 @@
+server {
+    listen *:80;
+    server_name _;
+
+    location ~ \.php$ {
+        fastcgi_keep_conn on;
+        fastcgi_pass   %php.host%:%php.port%;
+        fastcgi_index  index.php;
+    #    fastcgi_param  SCRIPT_FILENAME $document_root$fastcgi_script_name;
```
Comment (Ocramius) [inline docker/NGINX/default.conf]: I will need to get back to this before `1.0.0`, as I don't want any possible path hijacking going on /cc @EvanDotPro

**Pull request merged**